File: src/main/java/com/smart/appsa/model/enums/PadraoLamina.java
package com.smart.appsa.model.enums;

import com.fasterxml.jackson.annotation.JsonCreator;

import jakarta.persistence.EnumeratedValue;

public enum PadraoLamina {
    NENHUM(0),
    CASA(1),
    NAVIO(2),
    ESTRELA(3);

    @EnumeratedValue
    Integer value;

    @JsonCreator
    public static PadraoLamina fromValue(String value) {
        Integer parsed = Integer.valueOf(value);
        for (PadraoLamina p : values()) {
            if (parsed.equals(p.value)) return p;
        }
        throw new IllegalArgumentException("PadraoLamina inválido: " + value);
    }

    private PadraoLamina(Integer value) {
        this.value = value;
    }

    public Integer getValue() { return value; }
}

File: frontend/planta-smart-4.0/src/domain/enums/PadraoLamina.ts
export enum PadraoLamina {
  Nenhum = 'nenhum',
  Casa = 'casa',
  Navio = 'navio',
  Estrela = 'estrela',
}

export const PadraoLaminaToInt: Record<PadraoLamina, number> = {
  [PadraoLamina.Nenhum]: 0,
  [PadraoLamina.Casa]: 1,
  [PadraoLamina.Navio]: 2,
  [PadraoLamina.Estrela]: 3,
};

export const PadraoLaminaStringToEnum: Record<string, PadraoLamina> = {
  NENHUM: PadraoLamina.Nenhum,
  CASA: PadraoLamina.Casa,
  NAVIO: PadraoLamina.Navio,
  ESTRELA: PadraoLamina.Estrela,
};
